src/Automation/SnapshotCommand.cs

```diff
@@ -35,28 +36,57 @@ public static ScanResults Execute(Config config, IScanTools scanTools)
             if (config.CustomUIAConfigPath != null)
                 scanTools.Actions.RegisterCustomUIAPropertiesFromConfig(config.CustomUIAConfigPath);
 
+            List<ScanResults> resultList = new List<ScanResults>();
+
             var rootElements = scanTools.TargetElementLocator.LocateRootElements(config.ProcessId);
 
-            return scanTools.Actions.Scan(rootElements?.First(), (element, elementId) =>
+            if (rootElements is null || !rootElements.Any())
+            {
+                return resultList;
+            }
+
+            int targetIndex = 1;
+
+            foreach (var rootElement in rootElements)
             {
-                return ProcessResults(element, elementId, config, scanTools);
-            });
+                resultList.Add(scanTools.Actions.Scan(rootElement, (element, elementId) =>
+                {
+                    return ProcessResults(element, elementId, config, scanTools, targetIndex++, rootElements.Count());
```

Comment:
Hypothetical case: The app has 3 roots, and an Exception is thrown while scanning the last one. The current code would write the output for the 2 successful scans, then the Exception comes much later in output. As a customer, would you prefer to have the "this test failed to complete" message at the top or bottom of the output?

Reply:
I have to be honest that I mostly used Axe.Windows in a interaction test context where I am using the C# API. In that case, I would just want to have the exception be thrown since then I know that something went wrong when it shouldn't. A possible problem I see when interacting with the exceptions (setting them as the first item,etc.) that this could make it possibly harder to see the issue. Given that the exceptions could be because of anything, I think just have the test execution be "natural" is fine.